### ISSUE TITLE
Invalid interpreter causes crash over the LSP/pyls

### ISSUE THREAD

**Issue by dalthviz** [spyder-ide/spyder#13205]
@goanpeca @ccordoba12 , while testing #13104 I ran into an error where if you pass an invalid interpreter seems like it is being used by the LSP (?)

This happens not only in this PR but also in master and 4.x, although in 4.x no error report raises (but you can see that under the hood something is trying to execute the path given). I tested passing as interpreter a path to an installer (GIMP installer in this case):

```python

  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pyls_jsonrpc\endpoint.py", line 113, in consume
    self._handle_request(message['id'], message['method'], message.get('params'))
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pyls_jsonrpc\endpoint.py", line 182, in _handle_request
    handler_result = handler(params)
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pyls_jsonrpc\dispatchers.py", line 23, in handler
    return method(**(params or {}))
  File "c:\users\daniel\documents\spyder\spyder\external-deps\python-language-server\pyls\python_ls.py", line 334, in m_text_document__document_symbol
    return self.document_symbols(textDocument['uri'])
  File "c:\users\daniel\documents\spyder\spyder\external-deps\python-language-server\pyls\python_ls.py", line 250, in document_symbols
    return flatten(self._hook('pyls_document_symbols', doc_uri))
  File "c:\users\daniel\documents\spyder\spyder\external-deps\python-language-server\pyls\python_ls.py", line 156, in _hook
    return hook_handlers(config=self.config, workspace=workspace, document=doc, **kwargs)
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pluggy\hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pluggy\manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pluggy\manager.py", line 337, in traced_hookexec
    return outcome.get_result()
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pluggy\callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pluggy\callers.py", line 52, in from_call
    result = func()
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pluggy\manager.py", line 335, in <lambda>
    outcome = _Result.from_call(lambda: oldcall(hook, hook_impls, kwargs))
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pluggy\manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pluggy\callers.py", line 208, in _multicall
    return outcome.get_result()
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pluggy\callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\pluggy\callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "c:\users\daniel\documents\spyder\spyder\external-deps\python-language-server\pyls\plugins\symbols.py", line 12, in pyls_document_symbols
    definitions = document.jedi_names(all_scopes=all_scopes)
  File "c:\users\daniel\documents\spyder\spyder\external-deps\python-language-server\pyls\workspace.py", line 221, in jedi_names
    script = self.jedi_script()
  File "c:\users\daniel\documents\spyder\spyder\external-deps\python-language-server\pyls\workspace.py", line 234, in jedi_script
    environment = self.get_enviroment(environment_path) if environment_path else None
  File "c:\users\daniel\documents\spyder\spyder\external-deps\python-language-server\pyls\workspace.py", line 259, in get_enviroment
    environment = jedi.api.environment.create_environment(path=environment_path, safe=False)
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\jedi\api\environment.py", line 376, in create_environment
    return Environment(path)
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\jedi\api\environment.py", line 67, in __init__
    self._get_subprocess()
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\site-packages\jedi\api\environment.py", line 80, in _get_subprocess
    exc))

jedi.api.environment.InvalidPythonEnvironment: Could not get version information for 'C:/Users/Daniel/Documents/Quansight/gimp-2.10.14-setup-3.exe': InternalError("The subprocess C:/Users/Daniel/Documents/Quansight/gimp-2.10.14-setup-3.exe has crashed (EOFError('Ran out of input'), stderr=).")
```

Maybe this was already an error with an issue but just in case...

_Originally posted by @dalthviz in https://github.com/spyder-ide/spyder/pull/13104#issuecomment-654554498_

**Comment by ccordoba12** [comment]
So is it possible to set an invalid executable as a Python interpreter?

**Comment by dalthviz** [comment]
I'm not sure, the dialog with a warning indicating that the interpreter is not valid pops-up but seems like this is being bypassed and the selected interpreter starts being use by the LSP.

**Comment by ccordoba12** [comment]
We're probably connecting the line edit that sets the interpreter directly to the PyLS, without waiting to check if the interpreter is valid or not.

I'll take a look a it to see if I can fix for 4.1.4. Thanks @dalthviz for noticing it!
